src/nvim/api/window.c

```diff
@@ -462,3 +464,105 @@ void nvim_win_set_hl_ns(Window window, Integer ns_id, Error *err)
   win->w_hl_needs_update = true;
   redraw_later(win, UPD_NOT_VALID);
 }
+
+/// Counts the number of screen lines occupied by a range of text in a given window.
+/// Works for lines that are off-screen and takes folds into account.
+///
+/// Diff filler or virtual lines above a line are counted as a part of that line,
+/// unless the line is on "start_row" and "start_vcol" is specified.
+///
+/// Diff filler or virtual lines below the last line are only added to the result
+/// if "end_row" is omitted.
+///
+/// Line indexing is similar to |nvim_buf_get_text()|.
+///
+/// @param window  Window handle, or 0 for current window.
+/// @param opts    Optional parameters:
+///                - start_row: Line index where the range starts, 0-based inclusive.
+///                             When omitted start at the very top.
+///                - end_row: Line index where the range ends, 0-based inclusive.
+///                           When omitted end at the very bottom.
+///                - start_vcol: Virtual column on "start_row" where the range starts,
+///                              0-based, rounded down to full screen lines.
+///                              When omitted count the whole line.
+///                - end_vcol: Virtual column on "end_row" where the range ends,
+///                            0-based, rounded up to full screen lines.
+///                            When omitted count the whole line.
+/// @return  The number of screen lines that the range of text occupy.
+///
+/// @see |virtcol()| for text width.
+Integer nvim_win_text_height(Window window, Dict(win_text_height) *opts, Error *err)
```

Comment:
Are we absolutely sure that Integer will be sufficient for the future?
```suggestion
Dict nvim_win_text_height(Window window, Dict(win_text_height) *opts, Error *err)
```

Reply:
I'd rather change the return value type to `Object`, because naming Dictionary keys is hard.